### PR TITLE
JDK-8321073: Defer policy of disabling annotation processing by default

### PR DESCRIPTION
**Pull request by jddarcy**
To allow more time for build tools to accommodate changes in javac's default annotation processing policy, disabling annotation processing by default will be deferred one release to JDK 23. 

This PR re-enables implicit annotation processing by default _for JDK 22_. It is an anti-delta of  JDK-8306819.

Please also review the CSR to restore the previous behavior for JDK 22:

https://bugs.openjdk.org/browse/JDK-8321321

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8321321](https://bugs.openjdk.org/browse/JDK-8321321) to be approved

### Issues
 * [JDK-8321073](https://bugs.openjdk.org/browse/JDK-8321073): Defer policy of disabling annotation processing by default (**Enhancement** - P3)
 * [JDK-8321321](https://bugs.openjdk.org/browse/JDK-8321321): Defer policy of disabling annotation processing by default (**CSR**)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [3a6d5abf](https://git.openjdk.org/jdk/pull/16988/files/3a6d5abff5147b25aa688e7d386b1d3db3fad00d)
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**) ⚠️ Review applies to [3a6d5abf](https://git.openjdk.org/jdk/pull/16988/files/3a6d5abff5147b25aa688e7d386b1d3db3fad00d)
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**) ⚠️ Review applies to [3a6d5abf](https://git.openjdk.org/jdk/pull/16988/files/3a6d5abff5147b25aa688e7d386b1d3db3fad00d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16988/head:pull/16988` \
`$ git checkout pull/16988`

Update a local copy of the PR: \
`$ git checkout pull/16988` \
`$ git pull https://git.openjdk.org/jdk.git pull/16988/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16988`

View PR using the GUI difftool: \
`$ git pr show -t 16988`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16988.diff">https://git.openjdk.org/jdk/pull/16988.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16988#issuecomment-1841911150)

### CODE REVIEW

**Review by vicente-romero-oracle — APPROVED** [review]
looks good

**Review by lahodaj — APPROVED** [review]
LGTM

File: test/langtools/tools/javac/6341866/T6341866.java [hunk 1]
```diff
@@ -56,8 +56,8 @@ public class T6341866 {
     enum ImplicitType {
         NONE(null),                     // don't use implicit compilation
         OPT_UNSET(null),                // implicit compilation, but no -implicit option
-        OPT_NONE("-implicit:none"),     // implicit compilation with -implicit:none
-        OPT_CLASS("-implicit:class");   // implicit compilation with -implicit:class
+        OPT_NONE("-implicit:none"),     // implicit compilation wiith -implicit:none
```
Comment (mcimadamore) [inline test/langtools/tools/javac/6341866/T6341866.java:59]: Can we fix the typo?

File: test/langtools/tools/javac/6341866/T6341866.java [hunk 2]
```diff
@@ -56,8 +56,8 @@ public class T6341866 {
...
+        OPT_CLASS("-implicit:class");   // implicit compilation wiith -implicit:class
```
Comment (mcimadamore) [inline test/langtools/tools/javac/6341866/T6341866.java:60]: Here too

**Review by mcimadamore — APPROVED** [review]
Looks good

**Comment by jddarcy** [comment]
FYI, PR to implement the change in annotation processing default policy in JDK 23 out for review:

   https://github.com/openjdk/jdk/pull/19331

see also discussion thread

https://mail.openjdk.org/pipermail/jdk-dev/2024-May/009028.html

**Pull request closed without merging**